cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -29,6 +36,19 @@ import java.util.*
  * All methods called within the [FlowLogic] sub-class need to be annotated with the @Suspendable annotation.
  */
 object PaybackFlow {
+    fun findMoney(serviceHub: ServiceHub, party: Party): StateAndRef<CashState> {
+        // looking for money that should be printed on ExampleFlow
+        // this is not the best way, but I use this way to exercise, I will do better in module 4
+        val criteria = QueryCriteria.VaultQueryCriteria(
+                contractStateTypes = setOf(CashState::class.java),
+                status = Vault.StateStatus.UNCONSUMED,
+                exactParticipants = listOf(party)
+        )
+
```

Comment:
**Very Important:** I mentioned this in my review of your first attempt; pagination when querying the vault is very important. See [here](https://docs.corda.net/docs/corda-os/4.4/api-vault-query.html#pagination).

Reply:
An example of how to iterate through the pages: https://github.com/corda/corda/blob/f1ebaa761b6f4ca2538e67e62779ba5d426c9e46/node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt#L1651-L1663

Reply:
You have a mistake in the below line 👇 you are saying `queryBy IOUState` instead of `queryBy CashState`.